libsolidity/inlineasm/AsmCodeGen.cpp

```diff
@@ -217,42 +228,14 @@ class CodeTransform: public boost::static_visitor<>
 		int height = m_state.assembly.deposit();
```

Comment:
Since all local variables should be allocated without empty slots in between, `height - 1` needs to contain the last local variable.  Do you want to add such a check, or am I wrong in my reasoning?